right_arm_controller:
  follow_joint_trajectory:
    type: "robot_controllers/FollowJointTrajectoryController"
    joints:
      - right_s0
      - right_s1
      - right_e0
      - right_e1
      - right_w0
      - right_w1
      - right_w2
    gravity_compensation:
      type: "robot_controllers/GravityCompensation"
      root: "torso"
      tip: "right_gripper_base"
      autostart: true


left_arm_controller:
  follow_joint_trajectory:
    type: "robot_controllers/FollowJointTrajectoryController"
    joints:
      - left_s0
      - left_s1
      - left_e0
      - left_e1
      - left_w0
      - left_w1
      - left_w2
    gravity_compensation:
      type: "robot_controllers/GravityCompensation"
      root: "torso"
      tip: "left_gripper_base"
      autostart: true

gripper_controller:
  gripper_action:
    type: "robot_controllers/ParallelGripperController"
    centering:
      p: 1000.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
gazebo:
  default_controllers:
    - "right_arm_controller/follow_joint_trajectory"
    - "left_arm_controller/follow_joint_trajectory"
    - "gripper_controller/gripper_action"
  l_gripper_finger_joint:
    position:
      p: 1000
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  l_gripper_r_finger_joint:
    position:
      p: 1000
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  r_gripper_finger_joint:
    position:
      p: 1000
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  r_gripper_r_finger_joint:
    position:
      p: 1000
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 0.0
      d: 0.0
      i: 0.0
      i_clamp: 0.0
  right_s0:      
    position:
      p: 700
      d: 100
      i: 0.01
      i_clamp: 0.0  
    velocity:
      p: 200.0
      d: 0.1
      i: 0.01
      i_clamp: 1.0
  right_s1:
    position:           
      p: 10000
      d: 100
      i: 100
      i_clamp: 0.0
    velocity:
      p: 800.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  right_e0:
    position:
      p: 4500
      d: 1
      i: 35
      i_clamp: 0.0
    velocity:
      p: 300.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  right_e1:  
    position:
      p: 5500
      d: 2
      i: 60
      i_clamp: 0.0
    velocity:   
      p: 500.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  right_w0:   
    position:
      p: 1000
      d: 0.01
      i: 30.0
      i_clamp: 0.0
    velocity:  
      p: 200.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  right_w1:  
    position:
      p: 900.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 100.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  right_w2:       
    position:
      p: 1000.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 100.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
#  left_s0:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 200.0
#      d: 0.0
#      i: 2.0
#      i_clamp: 1.0
#  left_s1:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 200.0
#      d: 0.0
#      i: 0.0
#      i_clamp: 0.0
#  left_e0:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 10.0
#      d: 0.0
#      i: 0.0
#      i_clamp: 0.0
#  left_e1:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 150.0
#      d: 0.0
#      i: 0.0
#      i_clamp: 0.0
#  left_w0:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 150.0
#      d: 0.0
#      i: 0.0
#      i_clamp: 0.0
#  left_w1:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 100.0
#      d: 0.0
#      i: 0.0
#      i_clamp: 0.0
#  left_w2:
#    position:
#      p: 100.0
#      d: 0.1
#      i: 0.0
#      i_clamp: 0.0
#    velocity:
#      p: 100.0
#      d: 0.0
#      i: 0.0
#      i_clamp: 0.0
  left_s0:
    position:
      p: 700
      d: 100
      i: 0.01
      i_clamp: 0.0
    velocity:
      p: 200.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  left_s1:
    position:
      p: 10000
      d: 100
      i: 100
      i_clamp: 0.0
    velocity:
      p: 800.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  left_e0:
    position:
      p: 4500
      d: 1
      i: 35
      i_clamp: 0.0
    velocity:
      p: 300.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  left_e1:
    position:
      p: 5500
      d: 2
      i: 60
      i_clamp: 0.0
    velocity:
      p: 500.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  left_w0:
    position:
      p: 1000
      d: 0.01
      i: 30.0
      i_clamp: 0.0
    velocity:
      p: 200.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  left_w1:
    position:
      p: 900.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 100.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0
  left_w2:
    position:
      p: 1000.0
      d: 0.01
      i: 0.1
      i_clamp: 0.0
    velocity:
      p: 100.0
      d: 0.1
      i: 0.01
      i_clamp: 0.0